### DISCUSSION TITLE
What is the purpose of normalization in the adstock function?

### DISCUSSION THREAD

**Discussion by saukang** [google/lightweight_mmm#79] [Q&A] [answered]
Hey all,

Just wondering what is the reason behind the normalization in the adstock model, can anyone please clarify the intuition? Is it because it will be easier for the model to predict due to scaled (normalized) adstock? 

![image](https://user-images.githubusercontent.com/71245948/193002530-5d964941-5909-41fb-a2d3-470dfb5c5ca8.png)

**Comment by pabloduque0** [comment, marked as answer]
Hello @saukang !

This is due to the lagging in the adstock, if we dont normalize the values quickly build up.

Here is what the adstock values look like for two random media channels with normalisation:
![image](https://user-images.githubusercontent.com/13575259/193078216-24d643d8-33d6-4c19-91ba-a8a65f2805a2.png)

And here is what they would look like if we dont normalise:
![image](https://user-images.githubusercontent.com/13575259/193078300-09342d4f-8758-4064-bdc4-3ca63526e613.png)

However we will expose the parameter and you will be able to choose whether you want to normalise or not.

Hope it helps!

**Reply by saukang** [reply]
Thanks for the explanation and also the excellent visualization! 

Was it because the geometric adstock function is an exponential function that results in the adstock building up quickly without normalising?

**Reply by pabloduque0** [reply]
It is cumulative, feel free to check [the calculation](https://github.com/google/lightweight_mmm/blob/main/lightweight_mmm/media_transforms.py#L63).

The exponent there refers to the saturation which is then applied after the adstock.

**Reply by saukang** [reply]
I see and now I got it! Thanks a lot!
